core/trino-main/src/main/java/io/trino/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -237,6 +237,12 @@ public synchronized void startLifespan(Lifespan lifespan, ConnectorPartitionHand
         scheduleGroups.put(lifespan, new ScheduleGroup(partitionHandle));
         whenFinishedOrNewLifespanAdded.set(null);
         whenFinishedOrNewLifespanAdded = SettableFuture.create();
+
+        // schedule a task for collecting dynamic filters in case some dependent split generator is waiting for them
```

Comment:
I suggest we are more direct with this statement.  Something like
```
Avoid deadlocks by immediately scheduling a task for collecting dynamic filters, because there can be task in other stages blocked waiting for the dynamic filters.
```